module_utils/argspec/base.py

```diff
@@ -0,0 +1,24 @@
+from ansible.module_utils.six import iteritems
+
+
+class ArgspecBase(object):
+
+    argument_spec = {}
+
+    def __init__(self, **kwargs):
+        self.values = {}
+
+        for key, value in iteritems(kwargs):
+            if key in self.argument_spec:
+                setattr(self, key, value)
+
+    def __getattr__(self, key):
```

Comment:
This method should raise an `AttributeError` if the request key is not in the dict.   Also, in otder to avoid recursive lookups, the key should be checked against `__dict__` not `self.values`.

Reply:
Attributes that begin with `_` should only be checked against the internal structure not the instance variable, imho